### ISSUE TITLE
UNAS: Differentiable Architecture Search Meets Reinforcement Learning

### ISSUE THREAD

**Issue by sile** [sile/readings#183]
URL
-----

https://arxiv.org/abs/1912.07651

Abstract
----------

> Neural architecture search (NAS) aims to discover network architectures with desired properties such as high accuracy or low latency. Recently, differentiable NAS (DNAS) has demonstrated promising results while maintaining a search cost orders of magnitude lower than reinforcement learning (RL) based NAS. However, DNAS models can only optimize differentiable loss functions in search, and they require an accurate differentiable approximation of non-differentiable criteria. In this work, we present UNAS, a unified framework for NAS, that encapsulates recent DNAS and RL-based approaches under one framework. Our framework brings the best of both worlds, and it enables us to search for architectures with both differentiable and non-differentiable criteria in one unified framework while maintaining a low search cost. Further, we introduce a new objective function for search based on the generalization gap that prevents the selection of architectures prone to overfitting. We present extensive experiments on the CIFAR-10, CIFAR-100 and ImageNet datasets and we perform search in two fundamentally different search spaces. We show that UNAS obtains the state-of-the-art average accuracy on all three datasets when compared to the architectures searched in the DARTS space. Moreover, we show that UNAS can find an efficient and accurate architecture in the ProxylessNAS search space, that outperforms existing MobileNetV2 based architectures.

Table of Contents
--------------------

- (1) Introduction
  - (1-1) Related Work
- (2) Background
- (3) Method
  - (3-1) Gradient Estimation
  - (3-2) Training Objective
- (4) Experiments in DARTS Search Space
  - (4-1) Factorized Cell Structure
  - (4-2) Comparison with the Previous Work
- (5) Experiments on Latency-based Search
- (6) Conclusions
- (A) Architecture Search Settings
  - (A-1) Search Space
  - (A-2) CIFAR-10 and CIFAR-100
  - (A-3) ImageNet
- (B) Architecture Evaluation Settings
  - (B-1) CIFAR-10 and CIFAR-100
  - (B-2) ImageNet
- (C) Best Cell Structures
- (D) Comparison with the Previous Work in DARTS Space
- (E) UNAS with ProxylessNAS Search Space
  - (E-1) Search Space
  - (E-2) Evaluation

BibTeX
------

```bibtex
@misc{vahdat2019unas,
    title={UNAS: Differentiable Architecture Search Meets Reinforcement Learning},
    author={Arash Vahdat and Arun Mallya and Ming-Yu Liu and Jan Kautz},
    year={2019},
    eprint={1912.07651},
    archivePrefix={arXiv},
    primaryClass={cs.LG}
}
```
